.github/actions/sanitize/action.yml

```diff
@@ -40,23 +54,152 @@ runs:
 
     # https://github.com/actions/setup-node/issues/899
     - name: Enable Corepack before setting up Node
+      if: ${{ inputs.test-grafbase-gateway == 'false' }}
       shell: bash
       run: corepack enable
 
     - name: Setup Node.js
+      if: ${{ inputs.test-grafbase-gateway == 'false' }}
       uses: actions/setup-node@v4
       with:
         node-version: 18
 
     - name: Install pnpm
+      if: ${{ inputs.test-grafbase-gateway == 'false' }}
       uses: pnpm/action-setup@v2
       id: pnpm-install
       with:
         version: 8
         run_install: false
 
-    - if: ${{ inputs.test == 'true' }}
-      name: cargo nextest
+    - name: Login to Docker Hub
+      if: ${{ inputs.test-engine == 'true' }}
+      uses: docker/login-action@v3
+      with:
+        username: ${{ inputs.dockerhub-username }}
+        password: ${{ inputs.dockerhub-token }}
+
+    - name: Start local databases
+      shell: bash
+      if: ${{ inputs.test-engine == 'true' }}
+      run: |
+        docker-compose -f engine/crates/integration-tests/docker-compose.yml up -d
+
+    - if: ${{ inputs.test-engine == 'true' }}
+      name: test engine
       shell: bash
       run: |
-        RUST_BACKTRACE=1 cargo nextest run --workspace --exclude integration-tests --profile ci
+        RUST_BACKTRACE=1 cargo nextest run --workspace \
+          --exclude federated-server \
```

Comment:
Why not just run tests for everything? Sounds like a pain to maintain all those `--exclude`, I doubt anyone will remember to exclude stuff. I don't even see why we would only test one of the other to be honest.